File: testdata/jennies/builders/constraints/GoBuilder/constraints/somestruct_builder_gen.go
package constraints

import (
	cog "github.com/grafana/cog/generated/cog"
)

var _ cog.Builder[SomeStruct] = (*SomeStructBuilder)(nil)

type SomeStructBuilder struct {
    internal *SomeStruct
    errors cog.BuildErrors
}

func NewSomeStructBuilder() *SomeStructBuilder {
	resource := NewSomeStruct()
	builder := &SomeStructBuilder{
		internal: resource,
		errors: make(cog.BuildErrors, 0),
	}

	return builder
}



func (builder *SomeStructBuilder) Build() (SomeStruct, error) {
	if err := builder.internal.Validate(); err != nil {
		return SomeStruct{}, err
	}

	return *builder.internal, cog.MakeBuildErrors("constraints.someStruct", builder.errors)
}

func (builder *SomeStructBuilder) Id(id uint64) *SomeStructBuilder {
    builder.internal.Id = id

    return builder
}

func (builder *SomeStructBuilder) Title(title string) *SomeStructBuilder {
    builder.internal.Title = title

    return builder
}

File: testdata/jennies/builders/constraints/TypescriptBuilder/src/constraints/someStructBuilder.gen.ts
import * as cog from '../cog';
import * as constraints from '../constraints';

export class SomeStructBuilder implements cog.Builder<constraints.SomeStruct> {
    protected readonly internal: constraints.SomeStruct;

    constructor() {
        this.internal = constraints.defaultSomeStruct();
    }

    /**
     * Builds the object.
     */
    build(): constraints.SomeStruct {
        return this.internal;
    }

    id(id: number): this {
        if (!(id >= 5)) {
            throw new Error("id must be >= 5");
        }
        if (!(id < 10)) {
            throw new Error("id must be < 10");
        }
        this.internal.id = id;
        return this;
    }

    title(title: string): this {
        if (!(title.length >= 1)) {
            throw new Error("title.length must be >= 1");
        }
        this.internal.title = title;
        return this;
    }
}

